languages/rust/src/query.rs

```diff
@@ -0,0 +1,282 @@
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+use crate::host::{FromPolar, Instance, ToPolar};
+use polar_core::types::Symbol as Name;
+use polar_core::types::*;
+
+impl Iterator for Query {
+    type Item = anyhow::Result<ResultSet>;
+    fn next(&mut self) -> Option<Self::Item> {
+        Query::next_result(self)
+    }
+}
+
+pub struct Query {
+    inner: polar_core::polar::Query,
+    calls: HashMap<u64, Box<dyn Iterator<Item = Arc<dyn crate::host::ToPolar>>>>,
+    host: Arc<Mutex<crate::host::Host>>,
+}
+
+impl Query {
+    pub fn new(inner: polar_core::polar::Query, host: Arc<Mutex<crate::host::Host>>) -> Self {
+        Self {
+            calls: HashMap::new(),
+            inner,
+            host,
+        }
+    }
+
+    pub fn next_result(&mut self) -> Option<anyhow::Result<ResultSet>> {
```

Comment:
What are we getting with this `anyhow` crate? We don't use it for error handling in the core. (This can be tabled since I know error handling is a big open item).

Reply:
Temporary convenience. 

https://github.com/osohq/oso/pull/391/files/33fe4fb3bedac9a9ff6b302ddba03a23e94e2cd9#r482598147